frontend/src/js/services/WorkspaceApi.ts

```diff
@@ -43,8 +44,50 @@ export function getWorkspacesMetadata(): Promise<WorkspaceMetadata[]> {
     return authFetch('/api/workspaces').then(res => res.json());
 }
 
-export function getWorkspace(id: string) {
-    return authFetch(`/api/workspaces/${id}`).then(res => res.json());
+export function getWorkspace(id: string): Promise<Workspace> {
+    return Promise.all([
+      authFetch(`/api/workspaces/${id}`).then(res => res.json()),
+      authFetch(`/api/workspaces/${id}/remoteJobs`).then(res => res.json())
+    ]).then(([workspace, remoteJobs]: [Workspace, RemoteIngest[]]) => {
+
+        const inProgressRemoteJobs = remoteJobs.reduce((acc, job) => ({
+            ...acc,
+            [job.parentFolderId]: [
+              ...(acc[job.parentFolderId] || []),
```

Comment:
will this still work if the remote job is in a folder several levels deep?

Reply:
yes, this `inProgressRemoteJobs` reduce is essentially a groupBy on `parentFolderId` from the flat list of remote  jobs from the API, then in the subsequent `mixinInProgressRemoteJobs` function it calls itself recursively all the way down the tree and uses the `inProgressRemoteJobs` as a look-up (rather than many many calls to `find`)